### PR TITLE
Close all aggregators when closing onHeapIncrementalIndex

### PR DESCRIPTION
**Pull request by freakyzoidberg**
following discussion on https://groups.google.com/forum/#!topic/druid-development/u4b-H-UD3Es

Aggregators close() method is not called when OnHeapIncrementalIndex tear down.

### CODE REVIEW

File: processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java [hunk 1]
```diff
@@ -257,6 +257,15 @@ private void doAggregate(
     rowContainer.set(null);
   }
 
+  private void closeAggregators(Map<Integer, Aggregator[]> intMap)
```
Comment (leventov) [inline processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java]: This method could use the `aggregators` field, without having a parameter.

File: processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java [hunk 2]
```diff
@@ -257,6 +257,15 @@ private void doAggregate(
...
+  {
+    for (Aggregator[] aggregators : intMap.values()) {
```
Comment (leventov) [inline processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java]: Please use `Closer`

Comment (freakyzoidberg) [inline processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java]: Wouldn't that require Aggregator to extend Closeable ?

Comment (leventov) [inline processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java]: @freakyzoidberg huh, I didn't realise it doesn't. This use-case is a good reason to make to extend `Closeable`.

**Pull request merged**